### PR TITLE
fix(build): Drop support for Node 0.10 and 0.12.

### PR DESCRIPTION
**Pull request by johnjbarton**
Updated docs.
Updated engines matcher in package.json.
Updated places where node-0.1-specific code was noted in comments.

### CODE REVIEW

**Comment by johnjbarton** [comment]
Looks like travis was already updated.

File: docs/intro/04-faq.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ The latest stable version from NPM (`npm install karma`). See [versioning] for m
 
 
 ### Which version of Node.js does Karma run with?
-Karma works on all LTS versions node in active maintenance state (see [LTS docs](https://github.com/nodejs/LTS/blob/master/README.md) for more info) as well as the latest stable version. That is **0.12.x**, **4.x**, **5.x** and **6.x** at this point. Additionally, Node **0.10** is currently supported.
+Karma works on all LTS versions node in active maintenance state (see [LTS docs](https://github.com/nodejs/LTS/blob/master/README.md) for more info) as well as the latest stable version. That is **4.x**, **5.x** and **6.x** at this point.
```
Comment (dignifiedquire) [inline docs/intro/04-faq.md]: can you add that we support `8` as well please

File: package.json [hunk 2]
```diff
@@ -428,7 +428,7 @@
     "karma": "./bin/karma"
   },
   "engines": {
-    "node": "0.10 || 0.12 || 4 || 5 || 6 || 7 || 8"
+    "node": "4 || 5 || 6 || 7 || 8"
```
Comment (dignifiedquire) [inline package.json]: lets change this to `>= 4` please

File: docs/intro/04-faq.md (hunk 1 above)
Comment (johnjbarton) [reply docs/intro/04-faq.md]: done

File: package.json (hunk 2 above)
Comment (johnjbarton) [reply package.json]: done

**Comment by johnjbarton** [comment]
Done, please take another look.

**Comment by dignifiedquire** [comment]
Thank you :octocat:

**Pull request merged**